packages/x-components/src/plugins/__tests__/x-plugin-mixin.spec.ts

```diff
@@ -24,6 +25,66 @@ describe('testing $x component API global mixin', () => {
     jest.clearAllMocks();
   });
 
+  describe('testing origin', () => {
+    it('includes "default" origin in the $x.emit metadata', () => {
+      const listener = jest.fn();
+
+      componentInstance.vm.$x.on('UserIsTypingAQuery', true).subscribe(listener);
+      componentInstance.vm.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill');
+
+      expect(listener).toHaveBeenCalledTimes(1);
+      expect(listener).toHaveBeenCalledWith({
+        eventPayload: 'So awesome, much quality, such skill',
+        metadata: { moduleName: null }
+      });
+    });
+
+    it('overrides the origin in the $x.emit metadata', () => {
+      const listener = jest.fn();
+
+      @Component({
+        template: `<div><slot/></div>`,
+        provide: {
+          origin: 'origin-test'
+        }
+      })
+      class Provider extends Vue {}
+
+      @Component({
+        mixins: [xComponentMixin(searchBoxXModule)]
```

Comment:
It is not necessary to create an XComponent to test this.
Maybe rename the component to `Injector`
And maybe with a simple template, the component remains less verbose:

```
      @Component({ template: '<div>{{ $origin }}</div>' })
      class Injector extends Vue {}
```